packages/block-editor/src/components/link-control/style.scss

```diff
@@ -72,9 +72,7 @@ $block-editor-link-control-number-of-actions: 1;
 		color: $gray-900;
 	}
 
-	input[type="text"],
-	// Specificity overide of URLInput defaults.
-	&.block-editor-url-input input[type="text"].block-editor-url-input__input {
+	input[type="text"] {
```

Comment:
Everything here seems unnecessary, no?

```diff
diff --git a/packages/block-editor/src/components/link-control/index.js b/packages/block-editor/src/components/link-control/index.js
index 1f678f154d7..dfcc1dc5343 100644
--- a/packages/block-editor/src/components/link-control/index.js
+++ b/packages/block-editor/src/components/link-control/index.js
@@ -378,13 +378,13 @@ function LinkControl( {
 						{ showTextControl && (
 							<TextControl
 								__nextHasNoMarginBottom
+								__next40pxDefaultSize
 								ref={ textInputRef }
 								className="block-editor-link-control__field block-editor-link-control__text-content"
 								label={ __( 'Text' ) }
 								value={ internalControlValue?.title }
 								onChange={ setInternalTextInputValue }
 								onKeyDown={ handleSubmitWithEnter }
-								size="__unstable-large"
 							/>
 						) }
 						<LinkControlSearchInput
diff --git a/packages/block-editor/src/components/link-control/style.scss b/packages/block-editor/src/components/link-control/style.scss
index c760bbad734..500810f9653 100644
--- a/packages/block-editor/src/components/link-control/style.scss
+++ b/packages/block-editor/src/components/link-control/style.scss
@@ -67,30 +67,6 @@ $block-editor-link-control-number-of-actions: 1;
 
 .block-editor-link-control__field {
 	margin: $grid-unit-20; // allow margin collapse for vertical spacing.
-
-	.components-base-control__label {
-		color: $gray-900;
-	}
-
-	input[type="text"] {
-		@include input-control;
-		display: block;
-		border: $border-width solid $gray-600;
-		height: $button-size-next-default-40px; // components do not properly support unstable-large yet.
-		margin: 0;
-		padding: $grid-unit-10 $button-size-next-default-40px $grid-unit-10 $grid-unit-20;
-		position: relative;
-		width: 100%;
-
-		.has-actions & {
-			padding-right: $grid-unit-20;
-		}
-	}
-	//Specificity override of URL input defaults
-	.components-input-control input[type="text"].components-input-control__input {
-		border: none;
-		width: 100%;
-	}
 }
 
 .block-editor-link-control__search-error {

```

Reply:
-
-	.components-base-control__label {
-		color: $gray-900;
-	}
-
-	input[type="text"] {
-		@include input-control;
-		display: block;
-		border: $border-width solid $gray-600;
-		height: $button-size-next-default-40px; // components do not properly support unstable-large yet.
-		margin: 0;
-		padding: $grid-unit-10 $button-size-next-default-40px $grid-unit-10 $grid-unit-20;
-		position: relative;
-		width: 100%;
-
-		.has-actions & {
-			padding-right: $grid-unit-20;
-		}
-	}
-	//Specificity override of URL input defaults
-	.components-input-control input[type="text"].components-input-control__input {
-		border: none;
-		width: 100%;
-	}

This part is being utilized by the social-links and Image Links input field, If I remove this, then the width of input field shrinks to negligible for these two.

Anyways, I will try to remove these lines too.